### PR TITLE
Add connections argument to @deleteRecord directive

### PR DESCRIPTION
**Pull request by zth**
This adds support for removing nodes from connections when using the `@deleteRecord` directive.

### CODE REVIEW

File: packages/relay-runtime/handlers/connection/MutationHandlers.js [hunk 1]
```diff
@@ -27,9 +27,37 @@ import type {
 const DeleteRecordHandler = {
   update: (store: RecordSourceProxy, payload: HandleFieldPayload) => {
     const record = store.get(payload.dataID);
+
     if (record != null) {
       const id = record.getValue(payload.fieldKey);
+
       if (typeof id === 'string') {
+        const {connections} = payload.handleArgs;
```
Comment (zth) [inline packages/relay-runtime/handlers/connection/MutationHandlers.js:35]: I'm stuck on `connections` not appearing here in `handleArgs`, even though I _think_ I'm doing the transform right for passing the arguments on to here. But yeah, it's not working so obviously I'm doing something wrong.

**Comment by josephsavona** [comment]
cc @kassens @tyao1

**Comment by n1ru4l** [comment]
Hey I already mentioned it in the [Twitter Thread a while back](https://twitter.com/___zth___/status/1291085643662217216) and I wonder whether something like this could become part of relay:

> An interesting approach would be to automatically remove a deleted record from all referenced lists + remove all records that reference the deleted record (in a non-nullable way) up to the next nullable field/list

**Comment by zth** [comment]
@tyao1 sounds good! I'll change this to its own directive.

**Comment by zth** [comment]
Quick update: I'll fix this as soon as https://github.com/facebook/relay/pull/3155 lands, as that touches a lot of the same things.

**Comment by zth** [comment]
Superseded by https://github.com/facebook/relay/pull/3177.

**Pull request closed without merging**